spec/models/user_spec.rb

```diff
@@ -157,40 +167,67 @@
     context "when mobile does exist" do
       it "should send pin to existing user" do
         user = create(:user, mobile: mobile)
-        expect(User).to receive(:find_by_mobile).with(mobile).and_return(user)
+        user_attributes["email"] = nil
+        expect(User).to receive(:find_user_by_mobile_or_email).with(mobile, nil).and_return(user)
         expect(user).to receive(:send_verification_pin)
         User.creation_with_auth(user_attributes, DONOR_APP)
       end
     end
 
+    context "when email does exist" do
```

Comment:
Would be good to have specs to validate that the pin codes are sent through the correct medium (email or sms)

e.g
```
test "users without mobile should receive the pin code by email" do
   ...
  expect(user).to receive(:send_email_pin)
  expect(user).not_to receive(:send_sms_pin)
end
```

Three cases :

- user only has an email
- user only has a mobile
- user has both (and therefore, we decide what the expected behaviour)

I know that time is of the essence, so we can add those later  : )

Reply:
Nope that's not a use case. If user enters mobile number then he should receive sms and if enters email_id then will receive email

Reply:
Got it 👍 

In that case we can verify the new implementation is working as expected by replacing `expect(user).to receive(:send_verification_pin)` with `expect(user).to receive(:send_email_pin)`

Reply:
Added this spec. :)